frontend/components/AgentWallet/FundAgent/useFundAgent.tsx

```diff
@@ -0,0 +1,123 @@
+import { sum } from 'lodash';
+import { useCallback, useMemo, useState } from 'react';
+import { useUnmount } from 'usehooks-ts';
+
+import { CHAIN_CONFIG } from '@/config/chains';
+import { TOKEN_CONFIG, TokenConfig } from '@/config/tokens';
+import { TokenSymbol, TokenSymbolMap } from '@/constants';
+import {
+  useBalanceContext,
+  useMasterBalances,
+  useRewardContext,
+  useService,
+  useServices,
+  useUsdAmounts,
+} from '@/hooks';
+import { asEvmChainDetails } from '@/utils';
+
+import { AvailableAsset } from '../types';
+
+export const useFundAgent = () => {
+  const { isLoading: isBalanceLoading } = useBalanceContext();
+  const { accruedServiceStakingRewards } = useRewardContext();
+  const {
+    masterEoaNativeBalance,
+    masterSafeNativeBalance,
+    masterSafeOlasBalance,
+    masterSafeErc20Balances,
+  } = useMasterBalances();
+  const {
+    isLoading: isServicesLoading,
+    selectedAgentConfig,
+    selectedService,
+  } = useServices();
+  const { isLoaded } = useService(selectedService?.service_config_id);
+
+  const { evmHomeChainId, middlewareHomeChainId } = selectedAgentConfig;
+
+  const [amountsToWithdraw, setAmountsToWithdraw] = useState<
```

Comment:
shouldn't we call it amountsToFund? to me it's a bit misleading, as we're confusing the purpose of moving funds